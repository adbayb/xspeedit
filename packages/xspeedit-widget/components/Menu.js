import React, { Component } from "react";
import PropTypes from "prop-types";

class Menu extends Component {
	static propTypes = {
		entries: PropTypes.array,
		renderEntry: PropTypes.func
	};
	static defaultProps = {
		entries: []
	};

	renderItems() {
		const { entries, renderEntry } = this.props;

		return entries.map(renderEntry);
	}

	render() {
		return (
			<nav>
				{this.renderItems()}
				<style jsx>{`
					nav {
						display: flex;
						flex: 1;
						background-color: #303034;
						box-shadow: 0 -4px 6px rgba(0, 0, 0, 0.2);
					}
				`}</style>
			</nav>
		);
	}
}

export default Menu;
