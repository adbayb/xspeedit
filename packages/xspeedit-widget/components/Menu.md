Preview:

```js
<div style={{ display: "flex", height: 100 }}>
	<Menu
		entries={[
			{ iconName: "main", view: "main" },
			{ iconName: "github", url: "https://github.com/adbayb/xspeedit" }
		]}
		renderEntry={({ iconName, view, url }) => (
			<MenuItem
				key={iconName}
				iconName={iconName}
				isActive={view === "main"}
				url={url}
				onClick={() => {}}
			/>
		)}
	/>
</div>;
```

**Example:**

```jsx static
import React from "react";

const Example = () => {
	const handleClick = (e) => {
		console.log("click", e);
	};

	return (
		<Menu
			entries={[
				{ iconName: "main", view: "main" },
				{ iconName: "github", url: "https://github.com/adbayb/xspeedit" }
			]}
			renderEntry={({ iconName, view, url }) => (
				<MenuItem
					key={iconName}
					iconName={iconName}
					isActive={view === "main"}
					url={url}
					onClick={handleClick}
				/>
			)}
		/>
	);
};
```
