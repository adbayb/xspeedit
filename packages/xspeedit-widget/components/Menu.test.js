import React from "react";
import Menu from "./Menu.js";

describe("Menu", () => {
	it("should render correctly", () => {
		// GIVEN: notre composant:
		const tree = shallow(
			<Menu
				entries={[
					{ iconName: "main", view: "main" },
					{ iconName: "github", url: "https://github.com/adbayb/xspeedit" }
				]}
				renderEntry={(entry, index) => <div key={index}> {entry.iconName} </div>}
			/>
		);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
