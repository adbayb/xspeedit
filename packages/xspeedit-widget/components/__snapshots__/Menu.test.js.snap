// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<nav
  className="jsx-138632900"
>
  <div
    key="0"
  >
     
    main
     
  </div>
  <div
    key="1"
  >
     
    github
     
  </div>
  <JSXStyle
    css="nav.jsx-138632900{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;background-color:#303034;box-shadow:0 -4px 6px rgba(0,0,0,0.2);}"
    styleId="138632900"
  />
</nav>
`;
